features/device-config/persistence/impl/src/main/java/org/opennms/features/deviceconfig/persistence/impl/DeviceConfigDaoImpl.java

```diff
@@ -229,6 +229,12 @@ public int getLatestConfigCountForEachInterface(String searchTerm, Set<DeviceCon
         return count;
     }
 
+    @Override
+    public List<DeviceConfig> getAllDeviceConfigsWithAnInterfaceId(Integer ipInterfaceId) {
+        return find("from DeviceConfig dc where dc.lastUpdated is not null AND dc.ipInterface.id = ? ORDER BY lastUpdated DESC",
```

Comment:
May just need `from DeviceConfig dc where dc.ipInterface.id = ?`, i.e. don't need to sort and also this would catch entries where backup has not yet occurred.

Reply:
don't need sort but all entries should be deleted including entries that are waiting for backup

Reply:
`lastUpdated` will be null for entries that are waiting for backup

Reply:
good catch, will take care of that!